pkg/gitutil/git.go

```diff
@@ -51,8 +51,11 @@ func update(destinationPath string) error {
 		return errors.Wrapf(err, "fetch index at %q failed", destinationPath)
 	}
 
-	err := exec(destinationPath, "reset", "--hard", "@{upstream}")
-	return errors.Wrapf(err, "reset index at %q failed", destinationPath)
+	if err := exec(destinationPath, "reset", "--hard", "@{upstream}"); err != nil {
+		return errors.Wrapf(err, "reset index at %q failed", destinationPath)
+	}
+	err := exec(destinationPath, "clean", "-xfd")
```

Comment:
Argh.. there are no long options for `-xd` (only `--force/-f`). So can you add an explanatory comment what the git command does?

Reply:
made it a separate function, I guess it's good enough now.

Reply:
there was no need to make a new function IMO. there doesn't seem to be splitting these, esp. since we aren't testing them.

Reply:
This is exactly what I thought (tests) but then did it anyway because it was somehow addressing two comments and we might add tests later.

Reply:
Should I merge them back or good to keep?

Reply:
Yeah might as well merge back.
that method is basically 3 execs.

Reply:
Merged.

Reply:
@corneliusweig since godoc is updated, is inline comment needed?

Reply:
@ferhatelmas The godoc should be enough :)